rllib/models/specs/tests/test_check_specs.py

```diff
@@ -0,0 +1,268 @@
+            raise ValueError(ONLY_ONE_KEY_ALLOWED)
+        return self.run(input_dict)
+
+    @override(AbstractInterfaceClass)
+    def _check_only_input(self, input_dict: Dict[str, Any]) -> Dict[str, Any]:
+        # check if there is any key other than input in the input_dict
+        if len(input_dict) > 1 or "input" not in input_dict:
+            raise ValueError(ONLY_ONE_KEY_ALLOWED)
+
+        out = self.run(input_dict)
+
+        # output can be anything since ther is no output_spec
+        return {"output": str(out)}
+
+    @override(AbstractInterfaceClass)
+    def _check_only_output(self, input_dict) -> Dict[str, Any]:
+        # there is no input spec, so we can pass anything
+        if "input" in input_dict:
+            raise ValueError(
+                "input_dict should not have `input` key in check_only_output"
+            )
+
+        return self.run({"input": input_dict["not_input"]})
+
+
+class IncorrectImplementation(CorrectImplementation):
+    @override(CorrectImplementation)
+    def run(self, input_dict) -> Dict[str, Any]:
+        output = str(input_dict["input"] * 2)
+        return {"output": output}
+
+
+class TestCheckSpecs(unittest.TestCase):
+    def test_check_input_and_output(self):
+
+        correct_module = CorrectImplementation()
+
+        output = correct_module.check_input_and_output({"input": 2})
+        # output should also match the output_spec
+        correct_module.output_spec().validate(NestedDict(output))
+
+        # this should raise an error saying that the `input` key is missing
+        self.assertRaises(
+            ValueError, lambda: correct_module.check_input_and_output({"not_input": 2})
+        )
+
+    def test_check_only_input(self):
+        correct_module = CorrectImplementation()
+        # this should not raise any error since input matches the input specs
+        output = correct_module.check_only_input({"input": 2})
+        # output can be anything since ther is no output_spec
+        self.assertRaises(
+            ValueError,
+            lambda: correct_module.output_spec().validate(NestedDict(output)),
+        )
+
+    def test_check_only_output(self):
+        correct_module = CorrectImplementation()
+        # this should not raise any error since input does not have to match input_spec
+        output = correct_module.check_only_output({"not_input": 2})
+        # output should match the output specs
+        correct_module.output_spec().validate(NestedDict(output))
+
+    def test_incorrect_implementation(self):
+        incorrect_module = IncorrectImplementation()
+        # this should raise an error saying that the output does not match the
+        # output_spec
+        self.assertRaises(
+            ValueError, lambda: incorrect_module.check_input_and_output({"input": 2})
+        )
+
+        # this should not raise an error because output is not forced to be checked
+        incorrect_module.check_only_input({"input": 2})
```

Comment:
Dumb question, why would the decorator itself not already complain when it's being instantiated b/c of the missing output check? In other words, why is it allowed to have an implementation that doesn't check both, in- and output?

Reply:
`incorrect_module`'s `run` implementation does not have the correct output type. Therefore those functions that enforce output type checking should raise an error and those that don't should just ignore the output spec enforcement. This is only determined when the function is actually run and not during the function decoration itself.  Implementation is only visible to the decorator when the function is invoked.

Reply:
Ah, yes, that makes sense! Thanks for clarifying.